js/lightbeam.js

```diff
@@ -1,9 +1,56 @@
+// transforms the object of nested objects 'websites' into a
+// usable format for d3
+function transformData(websites) {
+  const nodes = [];
+  const links = [];
+  const firstParties = new Set();
+  const thirdParties = new Set();
+  // make first party nodes
+  for (const firstParty in websites) {
+    firstParties.add(firstParty);
+    const node = {};
+    node.hostname = firstParty;
+    node.favicon = websites[firstParty].faviconUrl || '';
+    node.party = 'first';
+    nodes.push(node);
+    // make third party nodes and links
+    if (websites[firstParty].thirdPartyRequests) {
+      for (const thirdParty in websites[firstParty].thirdPartyRequests) {
```

Comment:
`thirdParty` is a hostname only; in fact, both `firstParty` and `thirdParty` that are part of the `for in` loops are strings. I did something along what you are suggesting!